### DISCUSSION TITLE
Repeater pass-through interface Options props to child fields

### DISCUSSION THREAD

**Discussion by enzolupia** [directus/directus#13236] [Feature Requests]
As described in https://github.com/directus/directus/issues/13212 it would be useful to extend the Repeater interface so that the following props are passed to the selected child fields Options' context.

- type — The type of the child field.
- collection — The collection name where the Repeater is being added.
- field — The key of the child field.
- primaryKey — The current child field's item's primary key.
- (optional) —  The key of the repeater field 

This props are usually passed to the interfaces Options (from ExtensionOptionsContext I assume) but this is not true when using an interface inside a Repeater.
As a result of this interfaces that require the standard props to work (for example think about an interface that allows to select the current collection fields or similar) cannot be added as field inside a Repeater.

Adding a pass-through of this props would allow custom interfaces that required them to be added as child Field of the Repeater and improve usability of custom interfaces.

Thanks

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
